Transport/Get-EXOMTLReport.ps1

```diff
@@ -0,0 +1,376 @@
+Helpful in troubleshooting message delivery issues.
+
+.PARAMETER MTLFile
+MTL File to process.
+
+.PARAMETER ReportPath
+Folder path for the output file.
+
+.PARAMETER MessageID
+MessageID of a message to parse if there is more than one in the MTL.
+
+.OUTPUTS
+Text File broken into sections that contain the output of the various data gathering run against the MTL.
+
+Default Output File:
+$PSScriptRoot\MTL_Report_<date>.txt
+
+.EXAMPLE
+.\Get-EXOMTLReport.ps1 -MTLPath C:\temp\MyMtl.csv -MessageID <123214124@myserver.com>
+
+Generates a report from the MyMtl.csv file of the message with ID <123214124@myserver.com>
+
+#>
+
+[CmdletBinding()]
+param (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $MTLFile,
+    [Parameter()]
+    [string]
+    $ReportPath = $PSScriptRoot,
+    [Parameter()]
+    [string]
+    $MessageID
+)
+
+. $PSScriptRoot\..\Shared\ScriptUpdateFunctions\Test-ScriptVersion.ps1
+
+### Utilities ###
+function Import-MTL {
+    [CmdletBinding()]
+    [OutputType([array])]
+    param (
+        # File path for MTL to import
+        [Parameter(Mandatory = $true)]
+        [string]
+        $FilePath
+    )
+
+    # Test the path of the MTL
+    if (!(Test-Path $FilePath)) {
+        Write-Error "Unable to find the specified file" -ErrorAction Stop
+    }
+
+    # Try to load the file with Unicode since we need to start somewhere.
+    $initial_mtl = Import-Csv $FilePath -Encoding Unicode
+
+    # If it is null then we need to try without Unicode
+    if ($null -eq $initial_mtl) {
+        Write-Output "Failed to Load as Unicode; trying normal load"
+        $initial_mtl = Import-Csv $FilePath
+        # If we still have nothing then log an error and fail
+        if ($null -eq $initial_mtl) {
+            Write-Error "Failed to load CSV" -ErrorAction Stop
+        }
+        # Need to know that we loaded without Unicode.
+        else {
+            Write-Output "Loaded CSV without Unicode"
+        }
+    } else {
+        Write-Output "Loaded MTL with Unicode"
+    }
+
+    # Making sure the MTL contains the fields we want.
+    if (!(Test-CSVData -CSV $initial_mtl -ColumnsToCheck "date_time_utc", "source_context", "connector_id", "source", "event_id", "message_id", "recipient_address", "recipient_status", "recipient_count", "related_recipient_address", "reference", "message_subject", "sender_address", "return_path", "message_info", "directionality", "custom_data")) {
+        Write-Error "MTL is missing one or more required fields." -ErrorAction Stop
+    } else { Write-Output "CSV Passed Validation" }
+
+    # Converting our strings into [DateTime]
+    Write-Output "Converting date_time_utc values"
+    for ($i = 0; $i -lt $initial_mtl.Count; $i++) {
+        try {
+            $initial_mtl[$i].date_time_utc = Get-Date($initial_mtl[$i].date_time_utc)
+        } catch {
+            Write-Error ("Problem converting date information: " + $Error) -ErrorAction Stop
+        }
+    }
+
+    return $initial_mtl
```

Comment:
`Write-Output` within this function is also being returned here with `$initial_mtl`

Reply:
Moved all write-output to write-information

Reply:
By default, `$InformationPreference` is set to `SilentlyContinue` so you need to make sure this is set correctly if you always want it to be displayed.

Reply:
Yep ... bounced off that one.  

Setting it Continue then back to what the customer previously had it set to.